pkg/config/validation/validation.go

```diff
@@ -1851,6 +1851,39 @@ var ValidateRequestAuthentication = registerValidateFunc("ValidateRequestAuthent
 		return nil, errs
 	})
 
+// ValidateJwks checks that the JWKS json is valid.
+// Much like with Envoy's validation, if any of the keys are valid, the entire set is considered valid.
+// A JWKS with no keys present would cause Envoy to reject the config, so that's not considered valid.
+func ValidateJwks(raw []byte) (errs error) {
+	jwks := struct {
+		Keys []json.RawMessage `json:"keys"`
+	}{}
+	if err := json.Unmarshal(raw, &jwks); err != nil {
+		return err
+	}
+	// empty keys can cause an invalid Envoy config
+	if len(jwks.Keys) == 0 {
+		return fmt.Errorf("missing keys in jwks: \"%s\"", string(raw))
+	}
+	keyErrs := []error{}
```

Comment:
I'm still not sure if it's safe to rely on the golang JWT library which does not behave the same as the Envoy c++ JWT library when parsing jwks, a mismatch could potentially cause production issues for some unlucky customers. 

Should we take a step back to only return error for the obvious invalid case? For example, return error only if `jwks.Keys == 0`?

@incfly @liminw @myidpt wdyt?

Reply:
@yangminzhu don you know of specific differences between the two?

Reply:
For what it's worth, after thinking about this a bit more, I've added my opinion on what I'd like to encompass with the validation logic [here](https://github.com/istio/istio/issues/35582#issuecomment-978118762)